flake.nix

```diff
@@ -18,23 +18,24 @@
       with pkgs;
       {
         devShells.default = mkShell {
+          nativeBuildInputs = lib.optionals (stdenv.isDarwin) [
+            pkgs.darwin.apple_sdk.frameworks.SystemConfiguration
+          ];
           buildInputs = [
             openssl
             pkg-config
-            libudev-zero
             jq
             (rust-bin.stable.latest.default.override {
               extensions = [ "rust-src" ];
               targets = [ "wasm32-unknown-unknown" ];
             })
-          ];
+          ] ++ lib.optionals (stdenv.isLinux) [libudev-zero];
           shellHook =
           ''
             echo "Using `nix --version`"
             alias stellar="cargo run --bin stellar --"
```

Comment:
The fact the flake modifies the local shell's aliases is surprising and I think an anti-pattern. Aliases are something personal, something the terminal user should be in control of, and dev environments shouldn't fiddle with them imo.
```suggestion
```